File: firebird/jruby/create/firebird_create.rb
#! /usr/bin/jruby
# -*- encoding: utf-8 -*-
#
#	firebird_create.rb
#
#					May/21/2012
# ---------------------------------------------------------------------
include Java
import org.firebirdsql.jdbc.FBDriver
import java.sql.DriverManager
import java.lang.System
#
#
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
load '/var/www/data_base/common/jruby_common/jruby_sql_manipulate.rb'
# ---------------------------------------------------------------------
def data_prepare_proc ()
	dict_aa={}

	dict_aa=dict_append_proc(dict_aa,"t3821","松山",91547,"2007-6-21")
	dict_aa=dict_append_proc(dict_aa,"t3822","今治",86251,"2007-7-14")
	dict_aa=dict_append_proc(dict_aa,"t3823","宇和島",37263,"2007-3-27")
	dict_aa=dict_append_proc(dict_aa,"t3824","八幡浜",24298,"2007-9-29")
	dict_aa=dict_append_proc(dict_aa,"t3825","新居浜",15371,"2007-10-16")
	dict_aa=dict_append_proc(dict_aa,"t3826","西条",19645,"2007-11-24")
	dict_aa=dict_append_proc(dict_aa,"t3827","大洲",18196,"2007-12-30")
	dict_aa=dict_append_proc(dict_aa,"t3828","伊予",47384,"2007-5-18")
	dict_aa=dict_append_proc(dict_aa,"t3829","西予",52847,"2007-7-21")

	return dict_aa
end
# ---------------------------------------------------------------------
puts	"*** 開始 ***"
#
str_connect = "jdbc:firebirdsql:localhost/3050:/var/tmp/firebird/cities.fdb"
#
conn = DriverManager.getConnection(str_connect,"sysdba", "tiger")
#
dict_aa=data_prepare_proc()
#
sql_drop_table_proc(conn)
sql_create_table_proc(conn)
#
dict_aa.each {|key,value|
	sql_insert_record_proc(conn,key,value['name'], \
		value['population'],value['date_mod'])
	}
#
sql_read_proc(conn)
conn.close()
#
puts	"*** 終了 ***"
#
# ---------------------------------------------------------------------

File: firebird/clojure/create/firebird_create.clj
; -----------------------------------------------------------------
;
;	firebird_create.clj
;
;					May/21/2012
;
; -----------------------------------------------------------------
(import '(java.util Date))
(import '(java.text SimpleDateFormat))
(use 'clojure.contrib.sql)
(use '[clojure.contrib.str-utils :only (str-join)])
; -----------------------------------------------------------------
(defn insert_proc []
    (insert-values :cities
        [:id :name :population :date_mod] ["t3821" "松山" 65723 "1970-8-21"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t3822" "今治" 24589 "1970-7-9"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t3823" "宇和島" 74571 "1970-3-15"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t3824" "八幡浜" 42629 "1970-11-21"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t3825" "新居浜" 36892 "1970-10-7"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t3826" "西条" 42836 "1970-7-12"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t3827" "大洲" 81254 "1970-2-24"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t3828" "伊予" 91567 "1970-5-21"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t3829" "西予" 73619 "1970-9-17"])
    )

; -----------------------------------------------------------------
(println "*** 開始 ***")

(let [db-host "localhost"
	db-name "/var/tmp/firebird/cities.fdb"]

(def db {:classname "org.firebirdsql.jdbc.FBDriver"
	:subprotocol "firebirdsql"
	:subname (str db-host ":" db-name)
	:user "sysdba"
	:password "tiger"})

(with-connection db
	(drop-table :cities)
	(create-table :cities
		[:id "varchar(10)" "PRIMARY KEY"]
		[:name "varchar(20)"]
		[:population :int]
		[:date_mod :date])
	(insert_proc)
      ))
(println "*** 終了 ***")
; -----------------------------------------------------------------
